### PR TITLE
Fix text not vertically centered

### PR DESCRIPTION
**Pull request by bohan-v**
## Problems

Single line text render doesn't centered vertically, please check the following screenshots.

_Red box around text is the bounds of the label._

testRichTextBaseLine@3x
testRichTextBaseLine@3x 2
testRichTextBaseLine@3x 3

⚠️ This issue also in the existing project example

Simulator Screenshot - iPhone 8 - 2023-08-21 at 16 13 04

## After fix

testRichTextBaseLine@3x
testRichTextBaseLine@3x 2
testRichTextBaseLine@3x 3

✅ This also fixes the issue in the existing project example

Simulator Screenshot - iPhone 8 - 2023-08-21 at 16 11 27

## What does this PR include
1. update project settings with recommended settings
2. change running build phase script only on debug configuration
3. updates all failing snapshot tests (it's falling before the change). Update them for future reference

### CODE REVIEW

File: Source/Classes/Drawing/Drawing.swift [hunk 1]
```diff
@@ -118,7 +118,7 @@ extension NantesLabel {
             CTLineGetTypographicBounds(line, nil, &descent, nil)
 
             let penOffset = CGFloat(CTLineGetPenOffsetForFlush(line, flushFactor, Double(rect.size.width)))
-            let yOffset = lineOrigin.y - descent - font.descender
+            let yOffset = lineOrigin.y
```
Comment (bohan-v) [inline Source/Classes/Drawing/Drawing.swift:121]: This is the code that causes the bug. UIFont.descender is always negative value. `descent` is a Core Text value that  does gives similar value with UIFont.descender. This cause it renders at higher position than expected.

This calculate isn't necessary

Comment (ngocholo) [reply Source/Classes/Drawing/Drawing.swift:121]: can we remove the `descent` variable and the `CTLineGetTypographicBounds` call then?

**Review by ngoleo — COMMENTED** [review]
LGTM, let's clean up `descent` and merge

File: Source/Classes/Drawing/Drawing.swift (hunk 1 above)
Comment (bohan-v) [reply Source/Classes/Drawing/Drawing.swift:121]: yeah! right, will do. thanks!

**Review by ngoleo — APPROVED** [review]

**Comment by Kaspik** [comment]
This PR breaks rendering of custom fonts. We use Proxima Nuova (for example) and all labels are now clipping bottom of the label.
Screenshot 2023-09-05 at 17 54 22

I'm aware of https://github.com/instacart/Nantes/pull/88 but still letting you know in case you wanna debug.

**Pull request merged**